dali/operators/generic/one_hot.cc

```diff
@@ -63,96 +61,34 @@ This value will be cast to the ``dtype`` type.)code", 0.f)
 dimension in the output layout. If no character is provided, the output layout will be
 empty.)code", nullptr);
 
-namespace {
-
-template<int ndims>
-bool is_scalar(TensorShape<ndims> shape) {
-  return volume(shape) == 1;
-}
+class OneHotCPU : public OneHot<CPUBackend> {
+ public:
+  explicit OneHotCPU(const OpSpec &spec)
+      : OneHot<CPUBackend>(spec) {}
 
-TensorShape<> determine_shape(TensorShape<> in_shape, int num_classes, int axis,
-                              int output_sample_dim) {
-  if (output_sample_dim == 1) {
-    return {num_classes};
-  }
-  axis = axis < 0 ? in_shape.sample_dim() : axis;
-  int outer_axes = axis;
-  int inner_axes = in_shape.sample_dim() - axis;
-  auto outer = in_shape.first(outer_axes);
-  auto inner = in_shape.last(inner_axes);
-  return shape_cat(shape_cat(outer, num_classes), inner);
-}
+  ~OneHotCPU() override = default;
+  DISABLE_COPY_MOVE_ASSIGN(OneHotCPU);
```

Comment:
This is not really necessary - the operator won't be movable or copiable because the base class isn't.

Reply:
ok, fix'd:)